### PR TITLE
CLDR-15626 Document currencyPattern alts alphaNextToNumber,noCurrency and currencyPatternAppendISO

### PR DESCRIPTION
**Pull request by pedberg-icu**
CLDR-15626

- [x] This PR completes the ticket.

### CODE REVIEW

**Review by srl295 — APPROVED** [review]
thanks!

File: docs/ldml/tr35-numbers.md [hunk 1]
```diff
@@ -427,14 +435,39 @@ In addition to a standard currency format, in which negative currency amounts mi
...
+            <pattern type="100000000000000" count="other" alt="alphaNextToNumber">¤ 000T</pattern>
         </currencyFormat>
     </currencyFormatLength>
 </currencyFormats>
 ```
 
+The `alt="alphaNextToNumber"` pattern, if available, should be used instead of the standard pattern when the currency symbol character closest to the numeric value has Unicode General Category L (letter). The `alt="alphaNextToNumber"` pattern is typically provided when the standard currency pattern does not have a space between currency symbol and numeric value; the alphaNextToNumber variant adds a non-breaking space if appropriate for the locale.
+
+The `alt="noCurrency"` pattern can be used when a currency-style format is desired but without the currency symbol. This sort of display may be used when formatting a large column of values all in the same currency, for example. For compact currency formats (`<currencyFormatLength type="short">`), the cpmpact decimal format (`<decimalFormatLength type="short">`) should be used if no `alt="noCurrency"` pattern is present (so the `alt="noCurrency"` pattern is typically not needed for compact currency formats).
```
Comment (sffc) [inline docs/ldml/tr35-numbers.md:463]:
```suggestion
The `alt="noCurrency"` pattern can be used when a currency-style format is desired but without the currency symbol. This sort of display may be used when formatting a large column of values all in the same currency, for example. For compact currency formats (`<currencyFormatLength type="short">`), the compact decimal format (`<decimalFormatLength type="short">`) should be used if no `alt="noCurrency"` pattern is present (so the `alt="noCurrency"` pattern is typically not needed for compact currency formats).
```

File: docs/ldml/tr35-numbers.md [hunk 2]
```diff
@@ -427,14 +435,39 @@ In addition to a standard currency format, in which negative currency amounts mi
...
+
+```xml
+<currencyPatternAppendISO>{0} ¤¤</currencyPatternAppendISO>
+```
+
+The `currencyPatternAppendISO` element provides a pattern that can be used to combine currency format that uses a currency symbol (¤) with the ISO 4217 3-letter code for the same currency (¤¤), to produce a result such as “$1,432.00 USD”. Most locales will not need to override the pattern provided in root, shown above.
```
Comment (sffc) [inline docs/ldml/tr35-numbers.md:469]: How do we specify when implementations should use this pattern? Making the pattern available is a step in the right direction, but it's not very useful if we don't specify how to use it. The main example is when the currency symbol contains the ISO code, like "CHF". Should implementations do an exact string match? But then how about cases without ambiguity, like "CA$"; seems CAD is not really necessary for that. Maybe they should be paired with the narrow symbol, but only if the symbol is not equal to the ISO code?

File: docs/ldml/tr35-numbers.md (hunk 1 above)
Comment (pedberg-icu) [reply docs/ldml/tr35-numbers.md]: Fixed

File: docs/ldml/tr35-numbers.md (hunk 2 above)
Comment (pedberg-icu) [reply docs/ldml/tr35-numbers.md]: @sffc OK, I updated with a suggested recommendation for when to use.

**Comment by pedberg-icu** [comment]
Note in the recent update I also added a Mods item for https://github.com/unicode-org/cldr/pull/2367 since while this PR was outstanding I could not add it there due to collision.

**Review by sffc — COMMENTED** [review]
LGTM

**Pull request merged**